drivers/usb/udc/udc_numaker.c

```diff
@@ -31,36 +29,77 @@ LOG_MODULE_REGISTER(udc_numaker, CONFIG_UDC_DRIVER_LOG_LEVEL);
  *        to generate necessary 48MHz.
  */
 
+BUILD_ASSERT(!(DT_HAS_COMPAT_STATUS_OKAY(nuvoton_numaker_usbd) &&
+	       DT_HAS_COMPAT_STATUS_OKAY(nuvoton_numaker_hsusbd)),
+	     "NOT SUPPORT USBD/HSUSBD simultaneously");
+
+#if DT_HAS_COMPAT_STATUS_OKAY(nuvoton_numaker_usbd)
+#define DT_DRV_COMPAT                   nuvoton_numaker_usbd
+#define NUMAKER_XUSBD_TYPE              USBD_T
+#define NUMAKER_XUSBD_EP_TYPE           USBD_EP_T
+#define NUMAKER_XUSBD_SPEED_IDX_DEFAULT 1
+#elif DT_HAS_COMPAT_STATUS_OKAY(nuvoton_numaker_hsusbd)
+#define DT_DRV_COMPAT                   nuvoton_numaker_hsusbd
+#define NUMAKER_XUSBD_TYPE              HSUSBD_T
+#define NUMAKER_XUSBD_EP_TYPE           HSUSBD_EP_T
+#define NUMAKER_XUSBD_SPEED_IDX_DEFAULT 2
+#endif
+
 /* For bus reset, keep 'SE0' (USB spec: SE0 >= 2.5 ms) */
-#define NUMAKER_USBD_BUS_RESET_DRV_SE0_US 3000
+#define NUMAKER_XUSBD_BUS_RESET_DRV_SE0_US 3000
```

Comment:
These renaming changes, NUMAKER_USBD->NUMAKER_XUSBD numaker_usbd->numaker_xusbd, does not really make sense as the driver still supports USB 1.1 controller. It only makes the review harder. Therefore please drop these changes.